src/ToricVarieties/AlgebraicCycles/attributes.jl

```diff
@@ -0,0 +1,218 @@
+
+julia> d = ToricDivisor(dP2, [1,2,3,4,5])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> ac = AlgebraicCycle(d)
+An algebraic cycle on a normal toric variety given by 6*V(x3)+1*V(e1)+7*V(e2)
+
+julia> polynomial(ac)
+6*x3 + e1 + 7*e2
+```
+"""
+polynomial(ac::AlgebraicCycle) = ac.p
+export polynomial
+
+
+@doc Markdown.doc"""
+    polynomial(ring::MPolyQuo, ac::AlgebraicCycle)
+
+Returns the polynomial in `ring` corresponding
+to the algebraic cycle `ac`.
+
+# Examples
+```jldoctest
+julia> dP2 = del_pezzo(2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> d = ToricDivisor(dP2, [1,2,3,4,5])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> ac = AlgebraicCycle(d)
+An algebraic cycle on a normal toric variety given by 6*V(x3)+1*V(e1)+7*V(e2)
+
+julia> R, _ = PolynomialRing(QQ, 5)
+(Multivariate Polynomial Ring in x1, x2, x3, x4, x5 over Rational Field, fmpq_mpoly[x1, x2, x3, x4, x5])
+
+julia> (x1,x2,x3,x4,x5) = gens(R)
+5-element Vector{fmpq_mpoly}:
+ x1
+ x2
+ x3
+ x4
+ x5
+
+julia> sr_and_linear_relation_ideal = ideal([x1*x3, x1*x5, x2*x4, x2*x5, x3*x4, x1 + x2 - x5, x2 + x3 - x4 - x5])
+ideal(x1*x3, x1*x5, x2*x4, x2*x5, x3*x4, x1 + x2 - x5, x2 + x3 - x4 - x5)
+
+julia> R_quo = quo(R, sr_and_linear_relation_ideal)[1]
+Quotient of Multivariate Polynomial Ring in x1, x2, x3, x4, x5 over Rational Field by ideal(x1*x3, x1*x5, x2*x4, x2*x5, x3*x4, x1 + x2 - x5, x2 + x3 - x4 - x5)
+
+julia> polynomial(R_quo, ac)
+6*x3 + x4 + 7*x5
+```
+"""
+function polynomial(ring::MPolyQuo, ac::AlgebraicCycle)
+    p = polynomial(ac)
+    if iszero(p)
+        return zero(ring)
+    end
+    coeffs = [k for k in coefficients(p.f)]
+    expos = matrix(ZZ,[k for k in exponent_vectors(p.f)])
+    indets = gens(ring)
+    monoms = [prod(indets[j]^expos[k,j] for j in 1:ncols(expos)) for k in 1:nrows(expos)]
+    return sum(coeffs[k]*monoms[k] for k in 1:length(monoms))
+end
+export polynomial
+
+
+@doc Markdown.doc"""
+    irreducible_components(ac::AlgebraicCycle)
+
+Returns the irreducible components of the algebraic cycle `ac`.
+
+# Examples
+```jldoctest
+julia> dP2 = del_pezzo(2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> d = ToricDivisor(dP2, [1,2,3,4,5])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> ac = AlgebraicCycle(d)
+An algebraic cycle on a normal toric variety given by 6*V(x3)+1*V(e1)+7*V(e2)
+
+julia> irreducible_components(ac)
+3-element Vector{AlgebraicCycle}:
+ An algebraic cycle on a normal toric variety given by 1*V(x3)
+ An algebraic cycle on a normal toric variety given by 1*V(e1)
+ An algebraic cycle on a normal toric variety given by 1*V(e2)
+```
+"""
+function irreducible_components(ac::AlgebraicCycle)::Vector{AlgebraicCycle}
+    if is_irreducible(ac)
+        [ac]
```

Comment:
This is essentially a no-op. Did you perhaps mean this?
```suggestion
        return [ac]
```

Reply:
Yes, this was meant. Thank you.

Reply:
but you did not change it?

Reply:
Strange. I believe I did, but seems the change was lost? I committed this change now.